File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/attivita/risorse/Prodotto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;
using AgronicaCoreModelsSTD.baseClass;
using AgronicaCoreModelsSTD.metaschema;
using AgronicaCoreModelsSTD.metaschema.utilizzi;

namespace AgronicaCoreModelsSTD.attivita.risorse
{
    
    public class Prodotto : BaseCodeDescr
    {
        public int elemCod { get; set; }  // x compatibilità app
        
        public string codice_alfanumerico { get; set; }
        
        public TipoRisorsa tipo { get; set; }
        /// <summary>
        /// VANNI
        /// </summary>
        public UnitaDiMisura unitaDiMisura { get; set; }

        public Specie specie { get; set; }
        public Varieta varieta { get; set; }
        public GruppoFinalita finalita { get; set; }
        public Regolamenti regolamento { get; set; }


        public Prodotto(int codice) : base(codice, "")
        {
        }

        public Prodotto(int codice, int elem_cod) : base(codice,"")
        {
            elemCod = elem_cod;
        }

        public Prodotto() : base()
        {

        }

    }

}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/attivita/risorse/Prodotto.ts
import { UnitaDiMisura } from '../../metaschema/UnitaDiMisura';
import { BaseCodeDescr } from '../../baseClass/baseCodeDescr';
import { TipoRisorsa } from './TipoRisorsa';
import {Varieta} from "../../metaschema/utilizzi/Varieta";
import {Specie} from "../../metaschema/utilizzi/Specie";
import {GruppoFinalita} from "../../metaschema/utilizzi/GruppoFinalita";
import {Regolamenti} from "../../metaschema/Regolamenti";

export class Prodotto extends BaseCodeDescr {
  elemCod: number;
  unitaDiMisura: UnitaDiMisura;
  codice_alfanumerico: string;
  tipo: TipoRisorsa;

  specie: Specie;
  varieta: Varieta;
  finalita: GruppoFinalita;
  regolamento: Regolamenti;

  constructor(codice: number, descrizione?: string) {
    super(codice, descrizione);
  }
}
